### DISCUSSION TITLE
When using iotDB, in the case of multithreading, the Session will be used for a period of time, and the connection will be lost .  (golang语言， 在使用iotDB时，多线程情况下，Session使用超过一定时间后会报错失去连接)

### DISCUSSION THREAD

**Discussion by mygithubcjm** [apache/iotdb#3709] [Q&A] [answered]
1、程序初始化时，会初始化iotDB连接，并将得到的Session 使用公共变量存储
![1628578051(1)](https://user-images.githubusercontent.com/38320453/128820997-eeaf3578-8bdd-47d6-bc43-f197ddcb5291.png)

2、程序再收到MQ推送过来的数据后会插入数据到iotDB
![1628578251(1)](https://user-images.githubusercontent.com/38320453/128821397-94b41860-7c53-473f-b6d4-dce3e5727fb8.png)

3、checkError方法中如果发现错误会中断协程，然后捕获错误，并打印，之后再重新创建Session
![1628578359(1)](https://user-images.githubusercontent.com/38320453/128821621-25b7e4d8-7a31-4b7d-9de0-820c657692ad.png)
![1628578390(1)](https://user-images.githubusercontent.com/38320453/128821684-d9c2b19c-28f9-419b-9ede-3c9098061979.png)

测试用例：
使用jmeter工具向MQ中发送数据，程序接收输入后入库。
压测模拟20个数据包/秒，每个数据包只有2个测点

日志报错信息：
![1628578556(1)](https://user-images.githubusercontent.com/38320453/128822041-0453efe5-a261-4f87-a2c9-b330d6797370.png)

**Comment by qiaojialin** [comment, marked as answer]
Hi, is the Session used by multi-threads concurrently?

**Reply by mygithubcjm** [reply]
yes，it is.
![image](https://user-images.githubusercontent.com/38320453/128833530-de0584d1-c4a4-41d5-ad98-a8d579aae917.png)

**Reply by qiaojialin** [reply]
The session is not thread-safe, try to add concurrency control.
